config/file_test.go

```diff
@@ -0,0 +1,157 @@
+	if err == nil {
+		t.Errorf("Error was not returned")
+	}
+
+	r := regexp.MustCompile(`(?m:config file not found)`)
+	if !r.MatchString(err.Error()) {
+		t.Errorf("Error not matched: %s", err.Error())
+	}
+}
+
+func Test_UpdateAuthConfig_Insert(t *testing.T) {
+	DefaultDir = "~/.openfaas_test"
+	DefaultFile = "test2.yml"
+	u := "admin"
+	p := "some pass"
+	gatewayURL := strings.TrimRight("http://openfaas.test/", "/")
+	UpdateAuthConfig(gatewayURL, u, p)
+
+	user, pass, err := LookupAuthConfig(gatewayURL)
+	if err != nil {
+		t.Errorf("got error %s", err.Error())
+	}
+
+	if user != u || pass != p {
+		t.Errorf("got user %s and pass %s, expected %s %s", user, pass, u, p)
+	}
+}
+
+func Test_UpdateAuthConfig_Update(t *testing.T) {
+	DefaultDir = "~/.openfaas_test"
+	DefaultFile = "test3.yml"
+	u := "admin"
+	p := "pass"
+	gatewayURL := strings.TrimRight("http://openfaas.test/", "/")
+	UpdateAuthConfig(gatewayURL, u, p)
+
+	user, pass, err := LookupAuthConfig(gatewayURL)
+	if err != nil {
+		t.Errorf("got error %s", err.Error())
+	}
+
+	if user != u || pass != p {
+		t.Errorf("got user %s and pass %s, expected %s %s", user, pass, u, p)
+	}
+
+	u = "admin2"
+	p = "pass2"
+	UpdateAuthConfig(gatewayURL, u, p)
+
+	user, pass, err = LookupAuthConfig(gatewayURL)
+	if err != nil {
+		t.Errorf("got error %s", err.Error())
+	}
+
+	if user != u || pass != p {
+		t.Errorf("got user %s and pass %s, expected %s %s", user, pass, u, p)
+	}
+}
+
+func Test_UpdateAuthConfig_InvaidGatewayURL(t *testing.T) {
+	gateway := "http//test.test"
+	err := UpdateAuthConfig(gateway, "a", "b")
+	if err == nil {
+		t.Errorf("Error was not returned")
+	}
+
+	r := regexp.MustCompile(`(?m:invalid gateway)`)
+	if !r.MatchString(err.Error()) {
+		t.Errorf("Error not matched: %s", err.Error())
+	}
+}
+
+func Test_UpdateAuthConfig_EmptyGatewayURL(t *testing.T) {
+	gateway := ""
+	err := UpdateAuthConfig(gateway, "a", "b")
+	if err == nil {
+		t.Errorf("Error was not returned")
+	}
+
+	r := regexp.MustCompile(`(?m:invalid gateway)`)
+	if !r.MatchString(err.Error()) {
+		t.Errorf("Error not matched: %s", err.Error())
+	}
+}
+
+func Test_UpdateAuthConfig_EmptyUsername(t *testing.T) {
+	err := UpdateAuthConfig("http://test.test", "", "b")
+	if err == nil {
+		t.Errorf("Error was not returned")
+	}
+
+	r := regexp.MustCompile(`(?m:username)`)
+	if !r.MatchString(err.Error()) {
+		t.Errorf("Error not matched: %s", err.Error())
+	}
+}
+
+func Test_UpdateAuthConfig_EmptyPassword(t *testing.T) {
+	err := UpdateAuthConfig("http://test.test", "a", "")
+	if err == nil {
+		t.Errorf("Error was not returned")
+	}
+
+	r := regexp.MustCompile(`(?m:password)`)
+	if !r.MatchString(err.Error()) {
+		t.Errorf("Error not matched: %s", err.Error())
+	}
+}
+
+func Test_New_NoFile(t *testing.T) {
+	_, err := New("")
+	if err == nil {
+		t.Error("expected to fail on empty file path")
+	}
+}
+
+func Test_EnsureFile(t *testing.T) {
+	DefaultDir = "~/.openfaas_test"
```

Comment:
Can you use `ioutil.TempDir("", "faas-cli-file-test")` here so we don't pollute the users directory.